staging/src/k8s.io/endpointslice/topologies/manager.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topologies
+
+import (
+	"strings"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	discoveryv1 "k8s.io/api/discovery/v1"
+	"k8s.io/klog/v2"
+)
+
+// Manager exposes functions for working with multiple Topologies. It exposes
+// functions for adding and removing topology hints to endpoint slices using
+// various topologies.
+type Manager struct {
+	// topologiesByName maintains a mapping of a topology name to a Topology
+	// instance.
+	topologiesByName map[string]Topology
+	// defaultTopologyName is the name of the Topology which will be considered
+	// the default.
+	defaultTopologyName string
+
+	mu                        sync.Mutex
+	activeTopologiesByService map[string]map[discoveryv1.AddressType]Topology
+}
+
+// NewManager returns an instance of a topology Manager.
+//
+// `topologies` contains the set of topologies that this Mangaer will use and
+// manage. They should have unique names (as returned by their Topology.Names()
```

Comment:
I think we should be strict and normalize the names to require this to be case insensitive, so "TOPO1" == "topo1" == "Topo1"